FWCore/Services/plugins/InitRootHandlers.cc

```diff
@@ -346,7 +377,55 @@ namespace {
       return full_write(2, text);
     }
 
+    void sig_pause_for_stacktrace(int sig, siginfo_t*, void*) {
+      char buff[128] = "\nModule: ";
+ 
+      if (edm::CurrentModuleOnThread::getCurrentModuleOnThread() != nullptr) {
+        strlcat(buff, edm::CurrentModuleOnThread::getCurrentModuleOnThread()->moduleDescription()->moduleName().c_str(), 128);
+        full_cerr_write(buff);
+      }
+
+      struct timespec timeout = { edm::service::InitRootHandlers::stackTracePause(), 0 };
+#if defined(__linux__)
+      sigset_t sigset;
+      sigemptyset(&sigset);
+      sigaddset(&sigset, SIGCONT);
+      sigtimedwait(&sigset, NULL, &timeout);
+#else
+      nanosleep(&timeout); // macOS doesn't have sigtimedwait()
+#endif
+    }
+
     void sig_dostack_then_abort(int sig, siginfo_t*, void*) {
+      char buff[128] = "\nModule: ";
+
+      auto&& tids = edm::service::InitRootHandlers::threadIDs();
+
+      const auto self = pthread_self();
+#ifdef PAUSE_SIGNAL
+      if (edm::service::InitRootHandlers::stackTracePause() > 0 && tids.size() > 1) {
+        sigset_t pausesigset;
+        sigemptyset(&pausesigset);
+        sigaddset(&pausesigset,PAUSE_SIGNAL);
+        sigprocmask(SIG_UNBLOCK,&pausesigset,0);
+
+        int notified = 0;
+        for (auto id : tids) {
+          if (self != id) {
+            if (pthread_kill(id, PAUSE_SIGNAL) == 0) ++notified;
```

Comment:
Ah - from the latest man page (http://man7.org/linux/man-pages/man7/signal.7.html):
```
POSIX.1-2008 Technical Corrigendum 1 (2013) adds the following
       functions:

           fchdir()
           pthread_kill()
           pthread_self()
           pthread_sigmask()
```
So, I think we're safe here and above with `pthread_self`.